features/ui/button/button.tsx

```diff
@@ -18,3 +18,260 @@ export const Button = styled.button`
+          `
+        : props.color == "gray"
+        ? css`
+            background: #fff;
+            border: 1px solid ${color("gray", 300)};
+            color: ${color("gray", 700)};
+            box-shadow: ${shadow("xs")}, 0px 0px 0px 4px ${color("gray", 100)};
+          `
+        : props.color == "empty"
+        ? css`
+            color: ${color("primary", 700)};
+          `
+        : props.color == "emptyGray"
+        ? css`
+            color: ${color("gray", 500)};
+          `
+        : props.color == "error"
+        ? css`
+            background: ${color("error", 600)};
+            border: 1px solid ${color("error", 600)};
+            box-shadow: ${shadow("xs")}, 0px 0px 0px 4px ${color("error", 100)};
+          `
+        : css`
+            background: ${color("primary", 700)};
+            border: 1px solid ${color("primary", 700)};
+            box-shadow: ${shadow("xs")},
+              0px 0px 0px 4px ${color("primary", 100)};
+          `
+      : props.state == "disabled"
+      ? props.color == "secondary"
+        ? css`
+            background: ${color("primary", 25)};
+            border: 1px solid ${color("primary", 25)};
+            color: ${color("primary", 300)};
+            box-shadow: ${shadow("xs")};
+          `
+        : props.color == "gray"
+        ? css`
+            background: #fff;
+            border: 1px solid ${color("gray", 200)};
+            color: ${color("gray", 300)};
+            box-shadow: ${shadow("xs")};
+          `
+        : props.color == "empty"
+        ? css`
+            color: ${color("gray", 300)};
+          `
+        : props.color == "emptyGray"
+        ? css`
+            color: ${color("gray", 300)};
+          `
+        : props.color == "error"
+        ? css`
+            background: ${color("error", 200)};
+            border: 1px solid ${color("error", 200)};
+            box-shadow: ${shadow("xs")};
+          `
+        : css`
+            background: ${color("primary", 200)};
+            border: 1px solid ${color("primary", 200)};
+            box-shadow: ${shadow("xs")};
+          `
+      : null}
+
+${(props) => {
+    switch (props.icon) {
+      case ButtonIcon.only:
+        return css`
+          padding: 10px;
+        `;
+    }
+  }}
+`;
+
+export enum ButtonSize {
+  sm = "sm",
+  md = "md",
+  lg = "lg",
+  xl = "xl",
+}
+
+export enum ButtonColor {
+  primary = "primary",
+  secondary = "secondary",
+  gray = "gray",
+  empty = "empty",
+  emptyGray = "emptyGray",
+  error = "error",
+}
+
+export enum ButtonState {
+  default = "default",
+  hover = "hover",
+  focused = "focused",
+  disabled = "disabled",
+}
+
+export enum ButtonIcon {
+  none = "none",
+  leading = "leading",
+  trailing = "trailing",
+  only = "only",
+}
+
+type ButtonProps = {
+  children: React.ReactNode;
+  size?: ButtonSize;
+  color?: ButtonColor;
+  state?: ButtonState;
+  icon?: ButtonIcon;
+};
+
+export function StyledButton({
```

Comment:
I probably would have merged this with the original button. The original button could be the `empty` state or similar